2020/09/02/rclone-upload-to-cloud-from-cli/rclone-upload-to-cloud-from-cli.html.md

```diff
@@ -0,0 +1,239 @@
+
+This obviously makes uploading files from servers difficult- but not impossible, if you know the right tools.
+
+### Our use case
+
+Recently I had to put some large files in to long term storage on Google Drive, as we needed the space back. We wanted to retain the data, but didn't foresee needing to access it for some time, if ever. Google Drive was a good solution for us, however, the problem became how to get it there.
+
+They were too big, and some of them were not stored sparsely- empty space was tacked on to the disk images. There was also the issue of data integrity as the drives potentially contained customer information, so we wanted to encrypt them. There was also the issue of sequentially processing the files- encrypting them- then uploading them- I felt this would take quite a bit of time.
+
+Enter rclone. Rclone can connect to many different kinds of cloud storage providers, DIY cloud storage, and even things like FTP and Webdav. You can use rclone to copy files directly like rsync, or even use it to mount the remote storage as a local drive. The latter solution was what we chose to use.
+
+*Rclone can be used with a dizzying array of remote web services- including Dropbox, Box, Amazon S3, Mega, SugarSync, and even homebrew cloud like ownCloud! This example uses Google Drive, but the instructions for many cloud providers are similar- The setup wizard can guide you through each step.*
+
+We decided to mount the remote storage, then encrypt and compress the files with gpg outputting directly to the remote mount point, as this would allow us to set several different keys for decryption. Rclone offers a crypt module for encrypting backups, but this requires keeping a password- We prefer each person who may need to decrypt this be able to use a key. This also allowed me to run the command and walk away rather than try to shepherd the process along manually.
+
+Once the rclone mount was in place, encrypting the files for upload was very simple. Having the storage mounted locally negated the need to encrypt it locally and then upload it. We simply ran the following:
+
+```bash
+gpg --output /mnt/rclone/drive/sensitive_file.gpg --recipient recipient@email.com --encrypt ~/sensitive_file
+```
+
+Does this sound useful? Well, if it does...
+
+### How to do this yourself
+
+Installing rclone is as simple as downloading the binary from their website, https://rclone.org/, or cloning the git repository and compiling it yourself- They also offer a script to automatically download the most recent version. There doesn't seem to be a repository at this time.
```

Comment:
This seems contradictory, if I understand what you mean:

> cloning the git repository

vs.

> There doesn't seem to be a repository at this time.

Reply:
That was in reference to there not being a package repo- I've clarified that.